SimG4Core/Application/src/OscarMTMasterThread.cc

```diff
@@ -119,17 +103,35 @@ OscarMTMasterThread::~OscarMTMasterThread() {
   }
 }
 
+void OscarMTMasterThread::callConsumes(edm::ConsumesCollector&& iC) const {
+  if (m_hasToken) {
+    return;
+  }
+  if (m_pGeoFromDD4hep) {
+    m_DD4Hep = iC.esConsumes<cms::DDCompactView, IdealGeometryRecord, edm::Transition::BeginRun>();
+  } else {
+    m_DDD = iC.esConsumes<DDCompactView, IdealGeometryRecord, edm::Transition::BeginRun>();
+  }
+  m_PDT = iC.esConsumes<HepPDT::ParticleDataTable, PDTRecord, edm::Transition::BeginRun>();
+  m_hasToken = true;
+}
+
 void OscarMTMasterThread::beginRun(const edm::EventSetup& iSetup) const {
   std::lock_guard<std::mutex> lk(m_protectMutex);
-
   std::unique_lock<std::mutex> lk2(m_threadMutex);
+  edm::LogVerbatim("SimG4CoreApplication") << "OscarMTMasterThread::beginRun";
 
-  // Reading from ES must be done in the main (CMSSW) thread
-  readES(iSetup);
+  if (m_pGeoFromDD4hep) {
+    m_pDD4Hep = &(iSetup.getData(m_DD4Hep));
```

Comment:
To match what was done before
```suggestion
    m_pDD4Hep = &(*iSetup.getTransientHandle(m_DD4Hep));
```